src/neo/SmartContract/Native/ContractManagement.cs

```diff
@@ -63,6 +64,80 @@ internal ContractManagement()
             Manifest.Abi.Events = events.ToArray();
         }
 
+        private static void CheckScriptAndAbi(Script script, ContractAbi abi)
+        {
+            Dictionary<int, Instruction> instructions = new Dictionary<int, Instruction>();
+            for (int ip = 0; ip < script.Length;)
+            {
+                Instruction instruction = script.GetInstruction(ip);
+                instructions.Add(ip, instruction);
+                ip += instruction.Size;
+            }
+            foreach (var (ip, instruction) in instructions)
```

Comment:
In the past, I need to check if it's happens now, we used a `RET` and dummy data in order to randomize a transaction, with this, we will need to do it with a right opcodes.

Reply:
`RET` + `PUSHx`

Reply:
We have proper nonce in every transaction now, so I don't think it's needed. And these checks currently only affect deployed contracts, transactions can still have some garbage in their script (which is not good, but much less of an issue anyway).

Reply:
@roman-khimov Do you think we should check all the transactions and the witnesses?

Reply:
That's a tough question. Ideally, yes, but this can affect performance (unlike this PR which is almost free, contract deployments/updates are rare). At the same time scripts in transactions are usually short and we already parse witness verification scripts (`IsStandardContract` in `VerifyWitness`) without any noticeable issues. And this check is state-independent, which also is good (can be done concurrently).

So maybe it's possible. But if we're to do that I'd go with a bit more efficient checker.

Reply:
I've made a new PR to check all the transactions and witnesses. Please check #2266

Reply:
I think that it will be slower, we should trust in the vm

Reply:
For most transactions, only a few instructions need to be checked, so the speed will not be affected.